instructure/canvas-ios
# Replace Calendar Event Details TextEditor with RCE

refs: [MBL-17818](https://instructure.atlassian.net/browse/MBL-17818)
affects: Student, Teacher
release note: none

## What's new
- Replaced TextEditor with RCE for Add/Edit Calendar Event details
  - RCE should scroll to the end of it's content when focused
- Fix `ToggleCell` label swallowing taps

## What's not changed
Already existing issues with RCE:
- VoiceOver sometimes reads RCE content twice
- Links sometimes duplicate when edited
- RCE doesn't scroll to cursor position when focused

## Test plan
- Verify Add/Edit CalendarEvent Details RCE works in Student app
- Smoke test RCE in other places, eg.:
  - Submit text entry for assignment
  - Add announcement
  - Edit page

## Screenshots
<table>
<tr>
<td><img src="https://github.com/user-attachments/assets/b4627929-e6d1-42de-b11a-257d6ea771f6" maxHeight=500></td>
<td><img src="https://github.com/user-attachments/assets/3ad143a1-c730-4a37-ab6f-87d6adf0e917" maxHeight=500></td>
</tr>
</table>

## Checklist

[MBL-17818]: https://instructure.atlassian.net/browse/MBL-17818?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
